src/main/java/com/zendesk/maxwell/producer/MaxwellKinesisProducer.java

```diff
@@ -132,6 +132,12 @@ public void sendAsync(RowMap r, AbstractAsyncProducer.CallbackCompleter cc) thro
 		FutureCallback<UserRecordResult> callback = new KinesisCallback(cc, r.getNextPosition(), key, value,
 				this.succeededMessageCount, this.failedMessageCount, this.succeededMessageMeter, this.failedMessageMeter, this.context);
 
-		Futures.addCallback(future, callback);
+		try {
+			ListenableFuture<UserRecordResult> future = kinesisProducer.addUserRecord(kinesisStream, key, encodedValue);
+			Futures.addCallback(future, callback);
+		} catch(IllegalArgumentException t) {
+			callback.onFailure(t);
+			logger.error("Database:" + r.getDatabase() + ", Table:" + r.getTable() + ", PK:" + r.pkAsConcatString() + ", Size:" + Integer.toString(vsize));
```

Comment:
should we log the exception too?  Or is that going to be logged in `callback.onFailure`?

Reply:
`callback.onFailure` will log it. I was wondering whether to also log the beginning of the query (first 100 chars? 1000?) in case it wasn't an insert or update, but I'm not sure about that.